contracts/gateway/src/contract/execute.rs

```diff
@@ -102,3 +72,54 @@ pub fn route_outgoing_messages(
             .map(|msg| GatewayEvent::MessageRouted { msg }.into()),
     ))
 }
+
+fn partition_by_verified(
+    deps: DepsMut,
+    msgs: Vec<NewMessage>,
+) -> Result<(Vec<NewMessage>, Vec<NewMessage>), ContractError> {
+    let verifier = CONFIG.load(deps.storage)?.verifier;
```

Comment:
I'd still much prefer a pure execute mod without `deps`. To avoid repeated loading. In this case, `verifier` is already loaded in the function `verify_messages`.

Reply:
addressing this in a separate PR